### ISSUE TITLE
Spec fails 

### ISSUE THREAD

**Issue by padwasabimasala** [minimagick/minimagick#413]
When running `bundle` followed by `bundle exec rspec` one of the specs fails for "file not found"

```
MiniMagick::Configuration
  #configure
    yields self
  #cli
    can be assigned
    returns :imagemagick if #processor is mogrify
    returns :graphicsmagick if #processor is gm
    raises an error if #processor could not be found
  #cli=
    raises an error when set to an invalid value
  #processor
    assigns :mogrify by default
    assigns :gm if ImageMagick is not available
    returns nil if neither ImageMagick nor GraphicsMagick are available
  #processor=
    raises an error when set to an invalid value

With ImageMagick
  MiniMagick::Image
    has attributes (FAILED - 1)

Failures:

  1) With ImageMagick MiniMagick::Image has attributes
     Failure/Error: fail MiniMagick::Error, "`#{command.join(" ")}` failed with error:\n#{stderr}"

     MiniMagick::Error:
       `identify -format %m %w %h %b /var/folders/26/58rjss6d23zd3ktj90mnp4jw8rn327/T/mini_magick20170510-31051-izcxsc.jpg[0]` failed with error:
       identify: unable to open image `/var/folders/26/58rjss6d23zd3ktj90mnp4jw8rn327/T/mini_magick20170510-31051-izcxsc.jpg[0]': No such file or directory @ error/blob.c/OpenBlob/2705.
     # ./lib/mini_magick/shell.rb:17:in `run'
     # ./lib/mini_magick/tool.rb:92:in `call'
     # ./lib/mini_magick/tool.rb:40:in `new'
     # ./lib/mini_magick/image/info.rb:170:in `identify'
     # ./lib/mini_magick/image/info.rb:114:in `raw'
     # ./lib/mini_magick/image/info.rb:35:in `[]'
     # ./lib/mini_magick/image/info.rb:45:in `block in cheap_info'
     # ./lib/mini_magick/image/info.rb:44:in `fetch'
     # ./lib/mini_magick/image/info.rb:44:in `cheap_info'
     # ./lib/mini_magick/image/info.rb:17:in `[]'
     # ./lib/mini_magick/image.rb:127:in `block in attribute'
     # ./spec/lib/mini_magick/image_spec.rb:344:in `block (4 levels) in <top (required)>'

Finished in 0.10367 seconds (files took 1.23 seconds to load)
11 examples, 1 failure

Failed examples:

rspec ./spec/lib/mini_magick/image_spec.rb[1:1:12] # With ImageMagick MiniMagick::Image has attributes
```

**Comment by padwasabimasala** [comment]
Sorry to just dump this here. I spent a little time on it but wasn't able to find the solution.

```
$ convert -version
Version: ImageMagick 6.9.4-7 Q16 x86_64 2016-06-04 http://www.imagemagick.org
Copyright: Copyright (C) 1999-2016 ImageMagick Studio LLC
License: http://www.imagemagick.org/script/license.php
Features: Cipher DPC Modules
Delegates (built-in): bzlib freetype jng jpeg ltdl lzma png tiff xml zlib
$ ruby -v
ruby 2.2.2p95 (2015-04-13 revision 50295) [x86_64-darwin15]
```

**Comment by janko-m** [comment]
@padwasabimasala Could you try with the latest ImageMagick version? I've done some updates to the test suite once ImageMagick 7.x came out, and some of these changes might not be compatible with the 6.x versions.

**Comment by padwasabimasala** [comment]
Thanks for the reply. I upgraded and am getting a new error.

```
$ convert -version
Version: ImageMagick 7.0.5-5 Q16 x86_64 2017-04-25 http://www.imagemagick.org
...
```

```
With GraphicsMagick
  MiniMagick::Image
    has attributes (FAILED - 1)

Failures:

  1) With GraphicsMagick MiniMagick::Image has attributes
     Failure/Error: raise MiniMagick::Invalid, error.message

     MiniMagick::Invalid:
       `gm identify /var/folders/26/58rjss6d23zd3ktj90mnp4jw8rn327/T/mini_magick20170511-49866-7lvvbq.jpg` failed with error:
       executable not found: "gm"
     # ./lib/mini_magick/image.rb:203:in `rescue in validate!'
     # ./lib/mini_magick/image.rb:201:in `validate!'
     # ./lib/mini_magick/image.rb:113:in `block in create'
     # ./lib/mini_magick/image.rb:112:in `tap'
     # ./lib/mini_magick/image.rb:112:in `create'
     # ./lib/mini_magick/image.rb:34:in `read'
     # ./lib/mini_magick/image.rb:90:in `block in open'
     # ./lib/mini_magick/image.rb:89:in `open'
     # ./spec/lib/mini_magick/image_spec.rb:10:in `block (4 levels) in <top (required)>'
     # ./spec/lib/mini_magick/image_spec.rb:344:in `block (4 levels) in <top (required)>'
     # ------------------
     # --- Caused by: ---
     # MiniMagick::Error:
     #   `gm identify /var/folders/26/58rjss6d23zd3ktj90mnp4jw8rn327/T/mini_magick20170511-49866-7lvvbq.jpg` failed with error:
     #   executable not found: "gm"
     #   ./lib/mini_magick/shell.rb:17:in `run'

Finished in 4.48 seconds (files took 1.11 seconds to load)
112 examples, 1 failure

Failed examples:

rspec ./spec/lib/mini_magick/image_spec.rb[2:1:12] # With GraphicsMagick MiniMagick::Image has attributes```

**Comment by padwasabimasala** [comment]
When I was attempting to my original error yesterday, I ended up creating the one I just shared and am getting now.

It seemed to be related to layers and/or how file names are effected by layers.

While I was working on it yesterday I ran into some code that was appending "[0]" to the end of the file name in some cases. When I changed the code to remove "[0]" from the end of files names, it fixed the error I was getting, but created one similar to the one just shared.
